src/NServiceBus.AcceptanceTests/Core/OpenTelemetry/When_incoming_message_handled.cs

```diff
@@ -38,6 +47,13 @@ public async Task Should_record_failure_handling_time()
         Assert.AreEqual("failure", result);
     }
 
+    [Test]
+    public async Task Should_not_record_critical_time_on_failure()
+    {
+        using TestingMetricListener metricsListener = await WhenMessagesHandled(() => new MyExceptionalMessage());
+        metricsListener.AssertMetric(CriticalTimeMetricName, 0);
+    }
```

Comment:
@lailabougria, what do you think about this test? If that's what you meant, I'll add a similar one to the processing time metric PR